streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -92,7 +93,14 @@ int addRawRecords(final Iterable<ConsumerRecord<byte[], byte[]>> rawRecords) {
                 continue;
             }
 
-            final long timestamp = timestampExtractor.extract(record, timeTracker.get());
+            final long timestamp;
+            try {
+                timestamp = timestampExtractor.extract(record, timeTracker.get());
+            } catch (final StreamsException internalFatalExtractorException) {
+                throw internalFatalExtractorException;
```

Comment:
Could we either wrap the exception with additional context like topic/partition/offset, or make a log4j with these information? Ditto in line 102.

Reply:
We are the only ones that throw the `StreamsException` All the additional context is already added by the thrower, i.e., `FailOnInvalidTimestamp#onInvalidTimestamp`